### PR TITLE
Webpack: Expand version ranges of webpack in the apps (same as app/react)

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: Seems in some apps we have a direct dependency on webpack "4", when it should be a dependency on any version between 4 & 5 since we started supporting webpack 5.

@kroeder alerted me on dependency issues relating to angular, and I looked around the codebase and concluded this problem existed in a few more frameworks.

## What I did
I made the version of webpack we depend on a bigger range. This is exactly how we do it already for app/react.

### CODE REVIEW

**Comment by ndelangen** [comment]
@shilman perhaps something worthy of a patch?

**Comment by kroeder** [comment]
> @shilman perhaps something worthy of a patch?

I can try it in an alpha version first if you want to be extra sure! I'm using the latest @next version of Storybook

File: app/angular/package.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "ts-loader": "^8.0.14",
     "tsconfig-paths-webpack-plugin": "^3.3.0",
     "util-deprecate": "^1.0.2",
-    "webpack": "4"
+    "webpack": ">=4.0.0 <6.0.0"
```
Comment (shilman) [inline app/angular/package.json:83]: React's is a bit different. Do we care?
```
    "webpack": ">=4.43.0 <6.0.0"
```

**Review by shilman — APPROVED** [review]
LGTM!

File: app/angular/package.json (hunk 1 above)
Comment (ndelangen) [reply app/angular/package.json:83]: didn't want to introduce new constraints

**Comment by kroeder** [comment]
I tested the latest alpha that includes this PR and did a fresh re-install (deleted node_modules, package-lock.json, npm install)
I got no more errors in npm 8 🎉 

However, there are still webpack v4 versions deeply nested and npm does not want to hoist them

Here's the result of `npm ls webpack`

```
+-- @angular-devkit/build-angular@13.3.1
| +-- @angular-devkit/build-webpack@0.1303.1
| | `-- webpack@5.70.0 deduped
| +-- @ngtools/webpack@13.3.1
| | `-- webpack@5.70.0 deduped
| +-- babel-loader@8.2.3
| | `-- webpack@5.70.0 deduped
| +-- circular-dependency-plugin@5.2.2
| | `-- webpack@5.70.0 deduped
| +-- copy-webpack-plugin@10.2.1
| | `-- webpack@5.70.0 deduped
| +-- css-loader@6.5.1
| | `-- webpack@5.70.0 deduped
| +-- less-loader@10.2.0
| | `-- webpack@5.70.0 deduped
| +-- mini-css-extract-plugin@2.5.3
| | `-- webpack@5.70.0 deduped
| +-- postcss-loader@6.2.1
| | `-- webpack@5.70.0 deduped
| +-- sass-loader@12.4.0
| | `-- webpack@5.70.0 deduped
| +-- source-map-loader@3.0.1
| | `-- webpack@5.70.0 deduped
| +-- stylus-loader@6.2.0
| | `-- webpack@5.70.0 deduped
| +-- webpack-dev-middleware@5.3.0
| | `-- webpack@5.70.0 deduped
| +-- webpack-dev-server@4.7.3
| | `-- webpack@5.70.0 deduped
| +-- webpack-subresource-integrity@5.1.0
| | `-- webpack@5.70.0 deduped
| `-- webpack@5.70.0
+-- @storybook/addon-docs@6.5.0-alpha.53
| `-- @storybook/core-common@6.5.0-alpha.53
|   +-- fork-ts-checker-webpack-plugin@6.5.0
|   | `-- webpack@4.46.0 deduped
|   `-- webpack@4.46.0
|     `-- terser-webpack-plugin@1.4.5
|       `-- webpack@4.46.0 deduped
+-- @storybook/angular@6.5.0-alpha.53
| +-- @storybook/core@6.5.0-alpha.53
| | +-- @storybook/core-server@6.5.0-alpha.53
| | | +-- @storybook/builder-webpack4@6.5.0-alpha.53
| | | | +-- css-loader@3.6.0
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- file-loader@6.2.0
| | | | | `-- webpack@5.70.0 deduped
| | | | +-- html-webpack-plugin@4.5.2
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- postcss-loader@4.3.0
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- style-loader@1.3.0
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- terser-webpack-plugin@4.2.3
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- url-loader@4.1.1
| | | | | `-- webpack@5.70.0 deduped
| | | | +-- webpack-dev-middleware@3.7.3
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- webpack-filter-warnings-plugin@1.2.1
| | | | | `-- webpack@4.46.0 deduped
| | | | `-- webpack@4.46.0
| | | |   `-- terser-webpack-plugin@1.4.5
| | | |     `-- webpack@4.46.0 deduped
| | | +-- @storybook/manager-webpack4@6.5.0-alpha.53
| | | | +-- css-loader@3.6.0
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- html-webpack-plugin@4.5.2
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- style-loader@1.3.0
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- terser-webpack-plugin@4.2.3
| | | | | `-- webpack@4.46.0 deduped
| | | | +-- webpack-dev-middleware@3.7.3
| | | | | `-- webpack@4.46.0 deduped
| | | | `-- webpack@4.46.0
| | | |   `-- terser-webpack-plugin@1.4.5
| | | |     `-- webpack@4.46.0 deduped
| | | `-- webpack@4.46.0
| | |   `-- terser-webpack-plugin@1.4.5
| | |     `-- webpack@4.46.0 deduped
| | `-- webpack@5.70.0 deduped
| +-- postcss-loader@4.3.0
| | `-- webpack@5.70.0 deduped
| +-- raw-loader@4.0.2
| | `-- webpack@5.70.0 deduped
| +-- sass-loader@10.2.1
| | `-- webpack@5.70.0 deduped
| +-- ts-loader@8.3.0
| | `-- webpack@5.70.0 deduped
| `-- webpack@5.70.0 deduped
+-- @storybook/builder-webpack5@6.5.0-alpha.53
| +-- css-loader@5.2.7
| | `-- webpack@5.70.0 deduped
| +-- fork-ts-checker-webpack-plugin@6.5.0
| | `-- webpack@5.70.0 deduped
| +-- html-webpack-plugin@5.5.0
| | `-- webpack@5.70.0 deduped
| +-- style-loader@2.0.0
| | `-- webpack@5.70.0 deduped
| +-- terser-webpack-plugin@5.3.1
| | `-- webpack@5.70.0 deduped
| +-- webpack-dev-middleware@4.3.0
| | `-- webpack@5.70.0 deduped
| `-- webpack@5.70.0 deduped
+-- @storybook/manager-webpack5@6.5.0-alpha.53
| +-- @storybook/core-client@6.5.0-alpha.53
| | `-- webpack@5.70.0 deduped
| +-- css-loader@5.2.7
| | `-- webpack@5.70.0 deduped
| +-- webpack-dev-middleware@4.3.0
| | `-- webpack@5.70.0 deduped
| `-- webpack@5.70.0 deduped
+-- sass-loader@12.6.0
| `-- webpack@5.70.0 deduped
+-- storybook-addon-turbo-build@1.1.0
| `-- esbuild-loader@2.18.0
|   `-- webpack@5.70.0 deduped
`-- storybook-webpack-angular-types-plugin@0.2.0-0.hotfix-recursion.3
  `-- webpack@5.70.0 deduped
```

**Pull request merged**